ledger/src/blockstore_processor.rs

```diff
@@ -1406,7 +1412,9 @@ fn process_single_slot(
     })?;
 
     bank.freeze(); // all banks handled by this routine are created from complete slots
-    blockstore.insert_bank_hash(bank.slot(), bank.hash(), false);
+    if blockstore.is_primary_access() {
+        blockstore.insert_bank_hash(bank.slot(), bank.hash(), false);
```

Comment:
We may also move the is_primary_access check inside the insert_bank_hash to prevent the case where we forgot to check before inserting.